pex/compatibility.py

```diff
@@ -8,28 +8,35 @@
 
 import os
 from abc import ABCMeta
-from io import StringIO
 from sys import version_info as sys_version_info
 
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Optional
```

Comment:
This is the key part of the change, as explained in `pex/typing.py`.

Note that `from typing import TYPE_CHECKING` would fail on Py2 because `typing` is not defined.